File: php/Administrator.php
<?php

class Administrator extends User {
    private $system;
    private $requestsAssigned = [];
    private $requestsResolved = [];

    public function __construct($fname, $lname, $email) {
        parent::__construct($fname, $lname, $email, "Administrator");
    }

    public function assignRequest($request) {
        if (!$this->system->isRequestAssigned()) {
            $this->system->assignRequestToAdmin();
        }
        $this->requestsAssigned[] = $request;
    }

    public function viewRequestsAssigned() {
        foreach ($this->requestsAssigned as $request) {
            echo $request . "<br>";
        }
    }

    public function viewRequestsResolved() {
        foreach ($this->requestsResolved as $request) {
            echo $request . "<br>";
        }
    }

    public function assignPriority($request) {
        // Add your logic here
    }

    public function sendStatusUpdate() {
        // Add your logic here
    }

    public function resolveRequest($request) {
        $request->setRequestResolved(true);
        $this->requestsResolved[] = $request;
        $index = array_search($request, $this->requestsAssigned);
        if ($index !== false) {
            unset($this->requestsAssigned[$index]);
        }
        $this->system->sendRequestStatusUpdate();
    }
}

?>

File: Software Engineering/ELR Towers Class Structure/Administrator.java
import java.lang.reflect.Array;
import java.util.ArrayList;

public class Administrator extends User{
    ELRTowersDomesticAffairsSystem system;


        ArrayList<Request> requestsassigned = new ArrayList<Request>();
        ArrayList<Request> request_resolved = new ArrayList<Request>();

        public Administrator(String fname, String lname, String email){
            super(fname, lname, email, "Administrator");
        }
        public void assignRequest(Request request){
            for (Request r : requestsassigned){
                if(system.isRequestAssigned() == false){
                    system.assignRequestToAdmin();
                }
            requestsassigned.add(request);
        }
    }

        public void viewRequestsAssigned(){
            for (Request request : requestsassigned){
                System.out.println(request);
            }
        }


        public void viewRequestsResolved(){
            for (Request request : request_resolved){
                System.out.println(request);
            }
        }
        public void assign_priority(Request request){
           // continue;
        }
        public void sendStatusUpdate(){
            
        }

        public void resolveRequest(Request request){
            request. request_resolved = true;
            request_resolved.add(request);
            requestsassigned.remove(request);
            system.sendRequestStatusUpdate();
        }
        public void postNotice(Notice n){
            //placeholder, implementation to be given
        }

}
